part-customization.tex

```diff
@@ -50,18 +50,77 @@ \section{Introduction}
 \section{Workflow configuration}
 \label{sec-customization-workflow-configuration}
 
-LedgerSMB uses the concept of workflows\index{workflows} to manage the document life cycle.
-An invoice for example may be \texttt{saved}, \texttt{posted} or \texttt{reversed}.  These
-are states in the life cycle of the invoice.  Each state may have associated actions.  In
-case of the invoice an example may be to \texttt{Post} it when it's \texttt{saved} or to
-\texttt{E-mail} it when it's \texttt{posted}.
+LedgerSMB uses the concept of workflows\index{workflows} to manage object life cycle.
+An invoice, for example, may be \texttt{saved}, \texttt{posted} or \texttt{reversed}.  These
+are states in the life cycle of the invoice.  Each state may have associated actions. E.g. in
+case of the invoice there may be a \texttt{Post} action when it's \texttt{saved} or to
+\texttt{E-mail} it when it's \texttt{posted}.  Actions may cause the document to change state:
+if the \texttt{Post} action is executed on a \texttt{saved} invoice, its new state
+will be \texttt{posted}.  E-mailing an invoice does not change its state: it
+remains posted.  Instead, a new workflow is created which manages state of the e-mail.
+
+\digraph[scale=0.4]{wf1}{
+	rankdir=LR;
+	subgraph invoice {
+		graph [label="Invoice"];
+		cluster = true;
+		saved -> posted [label="post"];
+		posted -> reversed [label="reverse"];
+	};
+
+    subgraph email {
+    	graph [label="email"];
+    	cluster = true;
+    	created -> sent [label="send"];
+    };
+
+    posted -> created [label="e-mail"];
+    posted -> posted [label="e-mail"];
+
+}
+
+Actions belonging to a state may (or may not) be available. This is determined by one or
+more conditions.  Examples are ``is the posting date of the transaction in a closed period''
+and ``is the current user allowed to post transactions''.  When the condition evaluates to
+\texttt{false}, the action is not available.
+
+Workflows are configurable. Configurations list the following aspects:
+
+\begin{enumerate}
+	\item A list of states
+	\item A list of allowable actions per state
+	\item A list of conditions per action
+	\item A mapping of actions to Perl code
+	\item A mapping of conditions to Perl code
+\end{enumerate}
+
+With these ingredients, the software's behavior can be changed.  E.g. by adding a new state
+and action in the invoice, an extra reviewing pairs of eyes can be added to the posting process.
+By adding a condition on this action, it can be made required for invoices over 100.000 USD
+(but not for other invoices).
+
+\digraph[scale=0.4]{wf2}{

```

Comment:
This diagraph needs to be labelled as a Figure.

Reply:
Done.